packages/test/test-end-to-end-tests/src/test/gc/gcTrailingOps.spec.ts

```diff
@@ -0,0 +1,123 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+    ITestObjectProvider,
+    waitForContainerConnection,
+    createSummarizer,
+    summarizeNow,
+} from "@fluidframework/test-utils";
+import { describeNoCompat, ITestDataObject, TestDataObjectType } from "@fluidframework/test-version-utils";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { stringToBuffer } from "@fluidframework/common-utils";
+import { ISummaryTree } from "@fluidframework/protocol-definitions";
+import { getGCStateFromSummary } from "./gcTestSummaryUtils";
+import { defaultGCConfig } from "./gcTestConfigs";
+
+describeNoCompat("GC trailing ops tests", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+
+    /**
+     * Submits a summary and returns the unreferenced timestamp for all the nodes in the container. If a node is
+     * referenced, the unreferenced timestamp is undefined.
+     * @returns a map of nodeId to its unreferenced timestamp.
+     */
+    async function getUnreferencedTimestamps(summaryTree: ISummaryTree) {
+        const gcState = getGCStateFromSummary(summaryTree);
+        assert(gcState !== undefined, "GC tree is not available in the summary");
+        const nodeTimestamps: Map<string, number | undefined> = new Map();
+        for (const [nodeId, nodeData] of Object.entries(gcState.gcNodes)) {
+            nodeTimestamps.set(nodeId.slice(1), nodeData.unreferencedTimestampMs);
+        }
+        return nodeTimestamps;
+    }
+
+    beforeEach(async function() {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+    });
+
+    it("A summary has a datastore and blob referenced, but trailing ops unreferenced them", async () => {
```

Comment:
These tests are great!
You should add couple more tests where you wait for sweepTimeout before creating another summarizer with the summary. What that covers is scenarios where when the summarizer loads, the data store / blob will be tombstoned but the next summary would take into account the trailing ops and untombstone them.

Later when sweep is added, we will have to ensure that we don't delete objects before GC is run because trailing ops must be taken into account. Today, we mark them tombstoned before GC is run.